src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -1418,17 +1330,56 @@ private void UpdateBase2ExponentialHistogramWithMinMax(double number, ReadOnlySp
             histogram.RunningMax = Math.Max(histogram.RunningMax, number);
         }
 
-        if (reportExemplar && isSampled)
+        this.OfferExemplarIfSampled(number, tags, isSampled);
+
+        this.mpComponents.ReleaseLock();
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private readonly void OfferExemplarIfSampled<T>(T number, ReadOnlySpan<KeyValuePair<string, object?>> tags, bool isSampled)
+        where T : struct
+    {
+        if (isSampled)
         {
-            Debug.Assert(this.mpComponents.ExemplarReservoir != null, "ExemplarReservoir was null");
+            Debug.Assert(this.mpComponents?.ExemplarReservoir != null, "ExemplarReservoir was null");
```

Comment:
Why do we not use `this.mpComponents!` when we expect `this.mpComponents` to not be `null`? 

Reply:
We could do `this.mpComponents!`. The `OfferExemplarIfSampled` method isn't _technically_ validating `this.mpComponents != null` so having the `?` causes this one line to really validate that both `this.mpComponents` and `this.mpComponents.ExemplarReservoir` are not null. If someone messes up and starts calling this from a new path it might catch some mistake. We do know the outer methods today check `this.mpComponents` before calling into here so it is a little over-defensive. I'll change it if you want. But one note: `UpdateHistogram` (and friends) do the same type of thing (re-validate `this.mpComponents` even though it is checked before the call):

```csharp
Debug.Assert(this.mpComponents?.HistogramBuckets != null, "HistogramBuckets was null");
```